File: linked list.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None

    def insert_end(self, data):
        new_node = Node(data)

        if self.head is None:
            self.head = new_node
            return

        temp = self.head
        while temp.next:
            temp = temp.next

        temp.next = new_node

    def print_list(self):
        temp = self.head

        while temp:
            print(temp.data, end=" -> ")
            temp = temp.next

        print("None")


# Driver code
ll = LinkedList()

ll.insert_end(10)
ll.insert_end(20)
ll.insert_end(30)

ll.print_list()

File: linkedList.java
import java.util.LinkedList;

public class linkedList {
    Node head;
    static class Node{
        int data;
        Node next;
        Node(int data){
            this.data = data;
            this.next = null;
        }
    }
    public void insertNode(int data){
        Node newNode = new Node(data);
        if(head == null){
            head = newNode;   
        }
        else{
            Node t = head;
            while(t.next !=null){
                t = t.next;
                
            }
            t.next = newNode;
        }
    }

    public void display(){
        Node t = head;
        while(t != null){
            System.out.println(t.data + " ");
            t = t.next;
        }
    }

    public static void main(String[] args){
        linkedList List = new linkedList();
        List.insertNode(10);
        List.insertNode(20);
        List.insertNode(30);
        List.insertNode(40);
        List.insertNode(50);
        List.display();
    }
}


